enf-reducer/zero-constraint-subsumption.metta

```diff
@@ -0,0 +1,22 @@
+(= (zeroConstraintSubsume $expOriginal $expRecursive)
+(if (== $expRecursive ()) $expOriginal 
+(let*
+(
+    ($head (car-atom $expRecursive))
+    ($tail (cdr-atom $expRecursive))
+    ($headType (get-metatype $head))
+    ($headHasNoChild (not (nodeHasChildExp $head)))
+)
+(if (== $head AND) $expOriginal
```

Comment:
Isn't possible to use `case` here instead of nesting `if` expressions? It would make the code more readable.

Reply:
yeah that's right, I have corrected it as such